src/cursor/cursor.ts

```diff
@@ -205,13 +405,13 @@ export class Cursor<
     const currentNumberOfRetries = numberOfRetries;
 
     // Get the batchSize
-    let batchSize = 1000;
+    let batchSizeA = 1000;
```

Comment:
Why the rename to `batchSizeA`?  The `batchSize` pulled from `getLimitSkipBatchSizeDefaults` above seems at odds with this logic here now. Perhaps the one above should be the default, and you can omit the declaration here, only using the following logic to override the default if need be.